lang/csharp/src/apache/test/File/FileTests.cs

```diff
@@ -198,6 +199,45 @@ public void TestGenericData(string schemaStr, object[] value, Codec.Type codecTy
             }
         }
 
+        /// <summary>
+        /// This test is a single test case of the previous test but introduce a DeflateStream
+        /// as it is a standard non-seekable Stream that have the same behavior as the
+        /// NetworkStream that should be handled.
+        /// </summary>
+        [TestCase("{\"type\":\"record\", \"name\":\"n\", \"fields\":" +
+            "[{\"name\":\"f1\", \"type\":[\"int\", \"long\"]}]}",
+            new object[] { "f1", 100L }, Codec.Type.Null)]
+        public void TestNonSeekableStream(string schemaStr, object[] value, Codec.Type codecType)
+        {
+            foreach (var rwFactory in GenericOptions<GenericRecord>())
+            {
+                // Create and write out
+                MemoryStream compressedStream = new MemoryStream();
+                // using here a DeflateStream as it is a standard non-seekable stream, so if it works for this one,
+                // it should also works with any standard non-seekable stream (ie: NetworkStreams)
+                DeflateStream dataFileOutputStream = new DeflateStream(compressedStream, CompressionMode.Compress);
+                using (var writer = rwFactory.CreateWriter(dataFileOutputStream, Schema.Parse(schemaStr), Codec.CreateCodec(codecType)))
+                {
+                    writer.Append(mkRecord(value, Schema.Parse(schemaStr) as RecordSchema));
+                }
+
+                DeflateStream dataFileInputStream = new DeflateStream(new MemoryStream(compressedStream.ToArray()), CompressionMode.Decompress);
+
+                // Read back
+                IList<GenericRecord> readFoos = new List<GenericRecord>();
+                using (IFileReader<GenericRecord> reader = rwFactory.CreateReader(dataFileInputStream, null))
+                {
+                    foreach (GenericRecord foo in reader.NextEntries)
+                    {
+                        readFoos.Add(foo);
+                    }
```

Comment:
Ensure we throw an exception when calling one of the methods that requires a seekable stream.

```suggestion
                    }

                    // These methods are not supported for non-seekable streams.
                    Assert.Throws<AvroRuntimeException>(() => reader.Seek(0));
                    Assert.Throws<AvroRuntimeException>(() => reader.PreviousSync());
```